mlflow/sklearn.py

```diff
@@ -261,23 +261,44 @@ def log_model(sk_model, artifact_path, conda_env=None,
                      input_example=input_example)
 
 
-def _load_model_from_local_file(path):
-    """Load a scikit-learn model saved as an MLflow artifact on the local file system."""
+def _load_model_from_local_file(path, serialization_format):
+    """Load a scikit-learn model saved as an MLflow artifact on the local file system.
+
+    :param path: Local filesystem path to the MLflow Model saved with the ``sklearn`` flavor
+    :param serialization_format: The format in which the model was serialized. This should be one of
+                                 the following: ``mlflow.sklearn.SERIALIZATION_FORMAT_PICKLE`` or
+                                 ``mlflow.sklearn.SERIALIZATION_FORMAT_CLOUDPICKLE``.
+    """
     # TODO: we could validate the scikit-learn version here
+    if serialization_format not in SUPPORTED_SERIALIZATION_FORMATS:
+        raise MlflowException(
+                message=(
+                    "Unrecognized serialization format: {serialization_format}. Please specify one"
+                    " of the following supported formats: {supported_formats}.".format(
+                        serialization_format=serialization_format,
+                        supported_formats=SUPPORTED_SERIALIZATION_FORMATS)),
+                error_code=INVALID_PARAMETER_VALUE)
     with open(path, "rb") as f:
-        # Models serialized with Cloudpickle can be deserialized using Pickle; in fact,
-        # Cloudpickle.load() is just a redefinition of pickle.load(). Therefore, we do
-        # not need to check the serialization format of the model before deserializing.
-        return pickle.load(f)
+        # Models serialized with Cloudpickle cannot necessarily be deserialized using Pickle;
```

Comment:
Thanks for fixing this! This seems to be the crux of the issue (we should not have assumed that `pickle.load()` would continue to remain compatible with Cloudpickle as the package evolves).

Reply:
`cloudpickle` developer here. You are right, when `pickle5` is install ,`cloudpickle`-generated pickle strings cannot be unserialized using `pickle.load`. They **can** be unserialized using either

- `cloudpickle.load`
- `pickle5.load`

Just to clarify: `cloudpickle` did not deviate from the `pickle` API, it simply uses `pickle5` as a drop-in replacement of `pickle` (hence the compatibility with `pickle5.load)`. However, this is a breaking change and deserves at least a message in the `README`.

Reply:
@pierreglaser you're right, this deserves to be mentioned as it may cause issues for certain people.